templates/partials/contents.html

```diff
@@ -1,14 +1,23 @@
 {% for item in contents %}
+  {%- if item.definition -%}
+    {% set definition = item.definition %}
+    {% set definition_id = 'definition-' ~ definition.title.strip() | lower | replace(' ', '-') %}
```

Comment:
Not a fan of that at all, elements should be distinguishable, we should avoid indexes if at all possible. Indexes in tests are horrible and easily misread

Reply:
The id depends on the content of the title in this case which is not what we do with normal definitions where you can have any number of definitions. Same applies here, as you can have any number of definitions. Plus using the title will result in unnecessarily long IDs and func names.
If we don't want to use indexes because they are arrays, then each should have an "id" attribute.

Reply:
Normal question definitions are wrong, you shouldn't have id's which can't be determined. However I will changed it. Can you recommend how I can have a coherent index when I have other elements mixed in with contents like description

Reply:
Looks like you got it. It not the nicest thing but I think it is better than generating an id by manipulating user defined text. Ideally we would have an "id" attribute as part of the object.